### PR TITLE
[DOC] RDoc changes for IO

### PR DESCRIPTION
**Pull request by BurdetteLamar**
- Moves Expect library doc into `io.c`.
- Changes certain links to local sections, now pointing to sections in `doc/io_streams.rdoc`.
- Removes local sections now superseded by sections in `doc/io_streams.rdoc`.

### CODE REVIEW

File: io.c [hunk 1]
```diff
@@ -14390,183 +14393,47 @@ set_LAST_READ_LINE(VALUE val, ID _x, VALUE *_y)
...
- *  - <tt>'r+'</tt>: Read-write stream, positioned at the beginning.
- *  - <tt>'w+'</tt>: Read-write stream, positioned at the end.
- *  - <tt>'a+'</tt>: Read-write stream, positioned at the end.
- *
- *  For a writable file stream (that is, any except read-only),
- *  the file is truncated to zero if it exists,
- *  and is created if it does not exist.
- *
- *  Examples:
+ *  See IO#expect.
```
Comment (peterzhu2118) [inline io.c:14430]: Why don't we just move this part of the docs into the docs for the `IO#expect` method?

File: io.c [hunk 2]
```diff
@@ -14390,183 +14393,47 @@ set_LAST_READ_LINE(VALUE val, ID _x, VALUE *_y)
...
+ *
  *  - A position, which determines where in the stream the next
  *    read or write is to occur;
- *    see {Position}[rdoc-ref:IO@Position].
+ *    see {Position}[rdoc-ref:doc/io_streams.rdoc@Position].
  *  - A line number, which is a special, line-oriented, "position"
  *    (different from the position mentioned above);
- *    see {Line Number}[rdoc-ref:IO@Line+Number].
- *  - Internal and external encodings;
- *    see {Encodings}[rdoc-ref:IO@Encodings].
```
Comment (peterzhu2118) [inline io.c:14406]: The information about encodings isn't in the `io_streams.rdoc` file, so this information is lost.

**Review by peterzhu2118 — CHANGES_REQUESTED** [review]
There's a lot of broken links after this, for example, none of the `{Line Limit}[rdoc-ref:IO@Line+Limit]` or `{Line Separator and Line Limit}[rdoc-ref:IO@Line+Separator+and+Line+Limit]` links have been updated.

File: io.c (hunk 2 above)
Comment (BurdetteLamar) [reply io.c:14406]: There's a note and link to File a few lines above.

File: io.c (hunk 1 above)
Comment (BurdetteLamar) [reply io.c]: Done.

**Comment by BurdetteLamar** [comment]
> There's a lot of broken links after this, for example, none of the `{Line Limit}[rdoc-ref:IO@Line+Limit]` or `{Line Separator and Line Limit}[rdoc-ref:IO@Line+Separator+and+Line+Limit]` links have been updated.

Each link IO@ either corrected or verified as valid.

File: io.c [hunk 3]
```diff
@@ -2198,7 +2199,7 @@ rb_io_flush(VALUE io)
  *    tell -> integer
  *
  *  Returns the current position (in bytes) in +self+
- *  (see {Position}[rdoc-ref:IO@Position]):
+ *  (see {Position}[rdoc-ref:doc/io_streams.rdoc@Position]):
```
Comment (nobu) [inline io.c:2202]: This is a broken link.
```suggestion
 *  (see {Position}[rdoc-ref:io_streams.rdoc@Position]):
```

Comment (BurdetteLamar) [reply io.c:2202]: > This is a broken link, and all others.

Not broken on my machine.  I don't have make, so I run `rdoc --op html`, and have no breaks.  @peterzhu2118, how about you?  Or anyone else?

Comment (peterzhu2118) [reply io.c:2202]: This isn't broken for me either, but the suggested change from @nobu also works for me.

File: io.c (hunk 2 above)
Comment (peterzhu2118) [reply io.c:14406]: Ah I missed that.

**Review by peterzhu2118 — APPROVED** [review]
After we get @nobu's reply on broken links, I think it's good to merge!

File: io.c (hunk 3 above)
Comment (BurdetteLamar) [reply io.c]: Fixed (37 places).

**Comment by BurdetteLamar** [comment]
@nobu, ok to merge?

File: io.c (hunk 3 above)
Comment (BurdetteLamar) [reply io.c]: > This is a broken link, and all others.

These are fixed and verified.

**Pull request merged**